satellite/console/service.go

```diff
@@ -119,22 +119,41 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 		return nil, errs.New(internalErrMsg)
 	}
 
-	u, err = s.store.Users().Insert(ctx, &User{
-		Email:        user.Email,
-		FullName:     user.FullName,
-		ShortName:    user.ShortName,
-		PasswordHash: hash,
-	})
+	// store data
+	tx, err := s.store.BeginTx(ctx)
 	if err != nil {
-		return nil, errs.New(internalErrMsg)
+		return nil, err
 	}
 
-	err = s.store.RegistrationTokens().UpdateOwner(ctx, registrationToken.Secret, u.ID)
-	if err != nil {
-		return nil, errs.New(internalErrMsg)
-	}
+	// clean up user on error
+	defer func() {
+		if err != nil {
```

Comment:
this is pretty scary. this will suddenly start having bugs if for some reason someone starts changing the names of the return values. can you put this defer at the very top of the function before any errors are defined so it's clear we mean for it to refer to the named return value?